### DISCUSSION TITLE
GraphQL Int by default

### DISCUSSION THREAD

**Discussion by vasilev-alex** [MichalLytek/type-graphql#1117] [Q&A] [answered]
According to the docs https://typegraphql.com/docs/scalars.html#aliases when you don't provide `type` for a number it becomes a GraphQL `Float`. Is it possible to override this behavior to use GraphQL `Int` by default in this case?

**Comment by MichalLytek** [comment, marked as answer]
No, because `number` in JS is a double-precision floating point number value. You need to explicitly use it + figure out some TypeScript constructs to avoid runtime issues like returning `1.0000003` as record id or something 😉

**Reply by vasilev-alex** [reply]
Yeah, I see that. 

I just wanted to invert this behavior, because we have lots of `Int`s and few real `Float`s. 

```ts
export class Order {
  @Expose()
  // so this defaults to () => Int over () => Float
  @Field()
  id!: number;

  @Expose()
  // and if I actually need Float I explicitly pass the type
  @Field(() => Float)
  price!: number;
}
```

Do you think that makes sense?

**Reply by MichalLytek** [reply]
```ts
export const IntField = () => Field(() => Int);

export class Order {
  @IntField()
  id!: number;
}
```

**Reply by vasilev-alex** [reply]
That also works 😄
